src/payment/strategies/paypal-commerce/paypal-commerce-payment-processor.ts

```diff
@@ -0,0 +1,91 @@
+import { MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { Overlay } from '../../../common/overlay';
+
+const modalWidth = 450;
+const modalHeight = 600;
+
+export default class PaypalCommercePaymentProcessor {
+    private _window = window;
+    private _popup?: WindowProxy | null;
+    private _overlay: Overlay;
+
+    constructor() {
+        const innerHtml = this._getOverlayElements();
+        this._overlay = new Overlay({}, true, innerHtml);
+    }
+
+    paymentPayPal(approveUrl: string): Promise<boolean> {
+        return new Promise((resolve, reject) => {
+            const paramsWindow =  this._getParamsWindow();
+
+            const closeWindow = (isResolve: boolean, isRemoveOverlay: boolean = true) => {
+                this._window.removeEventListener('message', messageHandler);
+
+                if (this._popup) {
+                    this._popup.close();
+                    this._popup = undefined;
+                }
+
+                if (isRemoveOverlay) {
+                    this._overlay.remove();
+                }
+
+                isResolve
+                    ? resolve(true)
+                    : reject(new MissingDataError(MissingDataErrorType.MissingPayment));
+            };
+
+            const messageHandler = (event: MessageEvent) => {
+                if (event.origin === 'https://www.sandbox.paypal.com' || event.origin === 'https://www.paypal.com') {
+                    const data = JSON.parse(event.data);
+
+                    if (data.operation === 'return_to_merchant' && data.updateParent) {
+                        this._window.removeEventListener('message', messageHandler);
+                        closeWindow(true);
+                    }
+                }
+            };
+
+            this._window.addEventListener('message', messageHandler);
+            this._popup = this._window.open(approveUrl, 'PPFrame', paramsWindow);
+
+            const popupTick = setInterval(() => {
+                if (!this._popup || this._popup.closed) {
+                    clearInterval(popupTick);
+
+                    closeWindow(false);
+                }
+            }, 500);
+
+            this._overlay.show({
+                onClick: () => this._popup ? this._popup.focus() : closeWindow(false),
+                onClickClose: () => closeWindow(false, false),
+            });
+        });
+    }
+
+    private _getOverlayElements(): HTMLElement[] {
+        const text = document.createElement('div');
+        const link = document.createElement('a');
+
+        text.className = 'paypal-commerce-overlay_text';
+        text.innerText = 'Don\'t see the secure PayPal browser? We\'ll help you re-launch the window to complete your flow. You might need to enable pop-ups in your browser in order to continue.';
```

Comment:
We might want to pass in the text from the client application instead of hard-coding it inside the SDK (through `initialize` options). This is so that we can support non-English text.